### PR TITLE
Added labels following helm best practises

### PR DESCRIPTION
**Pull request by fjagugar**
### Description of the change

Adds labels to chart templates using the Helm best practises: https://helm.sh/docs/chart_best_practices/labels/#standard-labels

### Benefits

Tools already using the label syntax encouraged by Helm will now be able to use that syntax with kubapps.

### Possible drawbacks

Only added new tags. Old tags have not been removed for backward compatibility.

### CODE REVIEW

File: chart/kubeapps/templates/_helpers.tpl [hunk 1]
```diff
@@ -47,6 +49,7 @@ Labels to use on deploy.spec.selector.matchLabels and svc.spec.selector
 {{- define "kubeapps.matchLabels" -}}
 app: {{ include "kubeapps.name" . }}
 release: {{ .Release.Name }}
+app.kubernetes.io/instance: {{ .Release.Name }}
```
Comment (andresmgot) [inline chart/kubeapps/templates/_helpers.tpl:52]: here you are just adding one of the labels, is that intentional?

File: chart/kubeapps/templates/_helpers.tpl [hunk 2]
```diff
@@ -39,6 +39,8 @@ app: {{ include "kubeapps.name" . }}
 chart: {{ include "kubeapps.chart" . }}
 release: {{ .Release.Name }}
 heritage: {{ .Release.Service }}
+helm.sh/chart: {{ template "kubeapps.chart" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
```
[the remark this answers is not in the document]
Comment (fjagugar) [reply chart/kubeapps/templates/_helpers.tpl:43]: I see that not all the labels added in all the cases use the same parameters. It seem the `app/app.kubernetes.io/name` usually changes.

I've created a new template that adds all the labels that are common except those two ones and used it wherever those labels were the same.

**Comment by fjagugar** [comment]
I've created a diff on the template generated using `helm template` before and after the changes:

[template_differential.txt](https://github.com/kubeapps/kubeapps/files/4751359/template_differential.txt)

File: chart/kubeapps/templates/_helpers.tpl (hunk 2 above)
[the remark this answers is not in the document]
Comment (andresmgot) [reply chart/kubeapps/templates/_helpers.tpl:43]: :facepalm: this is a bit of a mess (our fault). I guess `app` should have been always the same but we changed it in several places and now we cannot fix it without making it a breaking change.

From the definition in the docs: `app.kubernetes.io/name: This should be the app name, reflecting the entire app. Usually {{ template "name" . }} is used for this. This is used by many Kubernetes manifests, and is not Helm-specific.`. Can we at least have a common `app.kubernetes.io/name`? We can move that new label to the common group.

File: chart/kubeapps/templates/_helpers.tpl [hunk 3]
```diff
@@ -47,6 +59,9 @@ Labels to use on deploy.spec.selector.matchLabels and svc.spec.selector
 {{- define "kubeapps.matchLabels" -}}
```
Comment (andresmgot) [inline chart/kubeapps/templates/_helpers.tpl:59]: I see this `matchLabels` is not used anywhere. Can we delete it?

**Comment by fjagugar** [comment]
New template diff generated:
[template_differential.txt](https://github.com/kubeapps/kubeapps/files/4787110/template_differential.txt)

**Pull request merged**